@rkothiya 
/home/jenkins/root/workspace/gh_centos7-regression/libglusterfs/src/dict.c: In function 'dict_del':
15:41:58 /home/jenkins/root/workspace/gh_centos7-regression/libglusterfs/src/dict.c:613:5: error: 'return' with a value, in function returning void [-Werror]
15:41:58      return dict_deln(this, key, strlen(key));
15:41:58      ^
15:41:58 /home/jenkins/root/workspace/gh_centos7-regression/libglusterfs/src/dict.c: In function 'dict_deln':
15:41:58 /home/jenkins/root/workspace/gh_centos7-regression/libglusterfs/src/dict.c:626:9: error: 'return' with no value, in function returning non-void [-Werror=return-type]
15:41:58          return;
15:41:58          ^
15:41:58 cc1: all warnings being treated as errors
15:41:58 make[3]: *** [libglusterfs_la-dict.lo] Error 1
15:41:58 make[3]: *** Waiting for unfinished jobs....
15:41:58 make[3]: Leaving directory `/d/build/scratch/libglusterfs/src'
15:41:58 make[2]: *** [install] Error 2
15:41:58 make[2]: Leaving directory `/d/build/scratch/libglusterfs/src'
15:41:58 make[1]: *** [install-recursive] Error 1
15:41:58 make[1]: Leaving directory `/d/build/scratch/libglusterfs'

regression is failing due to function void, kindly resolve it.

--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -619,11 +619,12 @@ dict_del(dict_t *this, char *key)
     return dict_deln(this, key, strlen(key));
 }
 
-void
+gf_boolean_t
 dict_deln(dict_t *this, char *key, const int keylen)
 {
     int hashval = 0;
     uint32_t hash;
+    gf_boolean_t rc = _gf_false;
 
     if (!this || !key) {
         gf_msg_callingfn("dict", GF_LOG_WARNING, EINVAL, LG_MSG_INVALID_ARG,
@@ -670,6 +671,7 @@ dict_deln(dict_t *this, char *key, const int keylen)
                 mem_put(pair);
             }
             this->count--;
+            rc = _gf_true;
             break;
         }
 
@@ -679,7 +681,7 @@ dict_deln(dict_t *this, char *key, const int keylen)
 
     UNLOCK(&this->lock);
 
-    return;
+    return rc;
 }
 
 void

--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -607,13 +607,13 @@ dict_key_count(dict_t *this)
     return ret;
 }
 
-void
+gf_boolean_t
 dict_del(dict_t *this, char *key)
 {
     if (!this || !key) {
         gf_msg_callingfn("dict", GF_LOG_WARNING, EINVAL, LG_MSG_INVALID_ARG,
                          "!this || key=%s", key);
-        return;
+        return _gf_false;
     }
 
     return dict_deln(this, key, strlen(key));
@@ -629,7 +629,7 @@ dict_deln(dict_t *this, char *key, const int keylen)
     if (!this || !key) {
         gf_msg_callingfn("dict", GF_LOG_WARNING, EINVAL, LG_MSG_INVALID_ARG,
                          "!this || key=%s", key);
-        return;
+        return rc;
     }
 
     hash = (uint32_t)XXH64(key, keylen, 0);

--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -149,7 +149,7 @@ data_t *
 dict_getn(dict_t *this, char *key, const int keylen);
 void
 dict_del(dict_t *this, char *key);
-void
+gf_boolean_t
 dict_deln(dict_t *this, char *key, const int keylen);
 int
 dict_reset(dict_t *dict);

--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -147,7 +147,7 @@ data_t *
 dict_get(dict_t *this, char *key);
 data_t *
 dict_getn(dict_t *this, char *key, const int keylen);
-void
+gf_boolean_t
 dict_del(dict_t *this, char *key);
 gf_boolean_t
 dict_deln(dict_t *this, char *key, const int keylen);

--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -213,6 +213,13 @@ enum gf_internal_fop_indicator {
 #define GF_XATTR_XTIME_PATTERN "trusted.glusterfs.*.xtime"
 #define GF_XATTR_TRIGGER_SYNC "glusterfs.geo-rep.trigger-sync"
 
+#define GLUSTERFS_ENTRYLK_COUNT_BIT 0
+#define GLUSTERFS_INODELK_COUNT_BIT 1
+#define GLUSTERFS_INODELK_DOM_COUNT_BIT 2
+#define GLUSTERFS_POSIXLK_COUNT_BIT 3
+#define GLUSTERFS_PARENT_ENTRYLK_BIT 4
+#define GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT 5
+
 /* quota xattrs */
 #define QUOTA_SIZE_KEY "trusted.glusterfs.quota.size"
 #define QUOTA_LIMIT_KEY "trusted.glusterfs.quota.limit-set"

--- a/xlators/features/locks/src/locks.h
+++ b/xlators/features/locks/src/locks.h
@@ -245,11 +245,7 @@ typedef struct {
     inode_t *inode;
     off_t offset;
     glusterfs_fop_t op;
-    gf_boolean_t entrylk_count_req;
-    gf_boolean_t inodelk_count_req;
-    gf_boolean_t posixlk_count_req;
-    gf_boolean_t parent_entrylk_req;
-    gf_boolean_t multiple_dom_lk_requests;
+    uint32_t bitfield;
     int update_mlock_enforced_flag;
 } pl_local_t;
 

--- a/xlators/features/locks/src/posix.c
+++ b/xlators/features/locks/src/posix.c
@@ -93,9 +93,14 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
             dict_unref(__unref);                                               \
     } while (0)
 
+#define IS_BIT_SET(byte, bit_number) (byte & (1 << bit_number))
+
+#define SET_BIT(byte, bit_number) (byte = (byte | (1 << bit_number)))
+
 #define PL_LOCAL_GET_REQUESTS(frame, this, xdata, __fd, __loc, __newloc)       \
     do {                                                                       \
-        if (pl_has_xdata_requests(xdata)) {                                    \
+        uint32_t bitfield = pl_has_xdata_requests(xdata);                      \
+        if (bitfield > 0) {                                                    \
             if (!frame->local)                                                 \
                 frame->local = mem_get0(this->local_pool);                     \
             pl_local_t *__local = frame->local;                                \
@@ -110,7 +115,7 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
                         loc_copy(&__local->loc[1], __newloc);                  \
                     __local->inode = inode_ref(__local->loc[0].inode);         \
                 }                                                              \
-                pl_get_xdata_requests(__local, xdata);                         \
+                pl_get_xdata_requests(__local, xdata, bitfield);               \
             }                                                                  \
         }                                                                      \
     } while (0)
@@ -171,7 +176,7 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
         __error;                                                               \
     })
 
-gf_boolean_t
+static uint32_t
 pl_has_xdata_requests(dict_t *xdata)
 {
     static char *reqs[] = {GLUSTERFS_ENTRYLK_COUNT,
@@ -189,15 +194,31 @@ pl_has_xdata_requests(dict_t *xdata)
                               SLEN(GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS),
                               0};
     int i = 0;
+    uint32_t bitfield = 0;
 
     if (!xdata)
         return _gf_false;
 
-    for (i = 0; reqs[i]; i++)
-        if (dict_getn(xdata, reqs[i], reqs_size[i]))
-            return _gf_true;
+    if (dict_del_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_ENTRYLK_COUNT_BIT);
 
-    return _gf_false;
+    if (dict_del_sizen(xdata, GLUSTERFS_INODELK_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_INODELK_COUNT_BIT);
+
+    if (dict_del_sizen(xdata, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS))
+        SET_BIT(bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT);
+
+    if (dict_del_sizen(xdata, GLUSTERFS_INODELK_DOM_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_INODELK_DOM_COUNT_BIT);
+
+    if (dict_getn(xdata, GLUSTERFS_POSIXLK_COUNT,
+                  SLEN(GLUSTERFS_POSIXLK_COUNT)))
+        SET_BIT(bitfield, GLUSTERFS_POSIXLK_COUNT_BIT);
+
+    if (dict_del_sizen(xdata, GLUSTERFS_PARENT_ENTRYLK))
+        SET_BIT(bitfield, GLUSTERFS_PARENT_ENTRYLK_BIT);
+
+    return bitfield;
 }
 
 static int
@@ -207,8 +228,8 @@ dict_delete_domain_key(dict_t *dict, char *key, data_t *value, void *data)
     return 0;
 }
 
-void
-pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
+static void
+pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, uint32_t bitfield)
 {
     if (!local || !xdata)
         return;
@@ -216,17 +237,7 @@ pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
     GF_ASSERT(local->xdata == NULL);
     local->xdata = dict_copy_with_ref(xdata, NULL);
 
-    if (dict_get_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT)) {
-        local->entrylk_count_req = 1;
-        dict_del_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT);
-    }
-    if (dict_get_sizen(xdata, GLUSTERFS_INODELK_COUNT)) {
-        local->inodelk_count_req = 1;
-        dict_del_sizen(xdata, GLUSTERFS_INODELK_COUNT);
-    }
-    if (dict_get_sizen(xdata, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS)) {
-        local->multiple_dom_lk_requests = 1;
-        dict_del_sizen(xdata, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS);
+    if (IS_BIT_SET(bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT)) {
         dict_foreach_fnmatch(xdata, GLUSTERFS_INODELK_DOM_PREFIX "*",
                              dict_delete_domain_key, NULL);
     }
@@ -235,17 +246,6 @@ pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
                                                   GLUSTERFS_INODELK_DOM_COUNT);
     if (local->inodelk_dom_count_req) {
         data_ref(local->inodelk_dom_count_req);
-        dict_del_sizen(xdata, GLUSTERFS_INODELK_DOM_COUNT);
-    }
-
-    if (dict_get_sizen(xdata, GLUSTERFS_POSIXLK_COUNT)) {
-        local->posixlk_count_req = 1;
-        dict_del_sizen(xdata, GLUSTERFS_POSIXLK_COUNT);
-    }
-
-    if (dict_get_sizen(xdata, GLUSTERFS_PARENT_ENTRYLK)) {
-        local->parent_entrylk_req = 1;
-        dict_del_sizen(xdata, GLUSTERFS_PARENT_ENTRYLK);
     }
 }
 
@@ -255,9 +255,7 @@ pl_needs_xdata_response(pl_local_t *local)
     if (!local)
         return _gf_false;
 
-    if (local->parent_entrylk_req || local->entrylk_count_req ||
-        local->inodelk_dom_count_req || local->inodelk_count_req ||
-        local->posixlk_count_req || local->multiple_dom_lk_requests)
+    if (local->bitfield)
         return _gf_true;
 
     return _gf_false;
@@ -538,13 +536,14 @@ pl_set_xdata_response(xlator_t *this, pl_local_t *local, inode_t *parent,
     if (!xdata || !local)
         return;
 
-    if (local->parent_entrylk_req && parent && name && name[0] != '\0')
+    if (IS_BIT_SET(local->bitfield, GLUSTERFS_PARENT_ENTRYLK_BIT) && parent &&
+        name && name[0] != '\0')
         pl_parent_entrylk_xattr_fill(this, parent, name, xdata, max_lock);
 
     if (!inode)
         return;
 
-    if (local->entrylk_count_req)
+    if (IS_BIT_SET(local->bitfield, GLUSTERFS_ENTRYLK_COUNT_BIT))
         pl_entrylk_xattr_fill(this, inode, xdata, max_lock);
 
     if (local->inodelk_dom_count_req)
@@ -552,13 +551,13 @@ pl_set_xdata_response(xlator_t *this, pl_local_t *local, inode_t *parent,
                               data_to_str(local->inodelk_dom_count_req),
                               max_lock);
 
-    if (local->inodelk_count_req)
+    if (IS_BIT_SET(local->bitfield, GLUSTERFS_INODELK_COUNT_BIT))
         pl_inodelk_xattr_fill(this, inode, xdata, NULL, max_lock);
 
-    if (local->posixlk_count_req)
+    if (IS_BIT_SET(local->bitfield, GLUSTERFS_POSIXLK_COUNT_BIT))
         pl_posixlk_xattr_fill(this, inode, xdata, max_lock);
 
-    if (local->multiple_dom_lk_requests)
+    if (IS_BIT_SET(local->bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT))
         pl_fill_multiple_dom_lk_requests(this, local, inode, xdata, max_lock);
 }
 

--- a/xlators/features/locks/src/posix.c
+++ b/xlators/features/locks/src/posix.c
@@ -38,6 +38,8 @@ int
 pl_lockinfo_get_brickname(xlator_t *, inode_t *, int32_t *);
 static int
 fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
+static void
+set_bit(uint32_t *byte, uint32_t bit_number);
 
 /*
  * The client is always requesting data, but older
@@ -95,8 +97,6 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
 
 #define IS_BIT_SET(byte, bit_number) (byte & (1 << bit_number))
 
-#define SET_BIT(byte, bit_number) (byte = (byte | (1 << bit_number)))
-
 #define PL_LOCAL_GET_REQUESTS(frame, this, xdata, __fd, __loc, __newloc)       \
     do {                                                                       \
         uint32_t bitfield = pl_has_xdata_requests(xdata);                      \
@@ -115,7 +115,8 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
                         loc_copy(&__local->loc[1], __newloc);                  \
                     __local->inode = inode_ref(__local->loc[0].inode);         \
                 }                                                              \
-                pl_get_xdata_requests(__local, xdata, bitfield);               \
+                __local->bitfield = bitfield;                                  \
+                pl_get_xdata_requests(__local, xdata);                         \
             }                                                                  \
         }                                                                      \
     } while (0)
@@ -176,47 +177,38 @@ fetch_pathinfo(xlator_t *, inode_t *, int32_t *, char **);
         __error;                                                               \
     })
 
+static void
+set_bit(uint32_t *byte, uint32_t bit_number)
+{
+    *byte = (*byte | (1 << bit_number));
+}
+
 static uint32_t
 pl_has_xdata_requests(dict_t *xdata)
 {
-    static char *reqs[] = {GLUSTERFS_ENTRYLK_COUNT,
-                           GLUSTERFS_INODELK_COUNT,
-                           GLUSTERFS_INODELK_DOM_COUNT,
-                           GLUSTERFS_POSIXLK_COUNT,
-                           GLUSTERFS_PARENT_ENTRYLK,
-                           GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS,
-                           NULL};
-    static int reqs_size[] = {SLEN(GLUSTERFS_ENTRYLK_COUNT),
-                              SLEN(GLUSTERFS_INODELK_COUNT),
-                              SLEN(GLUSTERFS_INODELK_DOM_COUNT),
-                              SLEN(GLUSTERFS_POSIXLK_COUNT),
-                              SLEN(GLUSTERFS_PARENT_ENTRYLK),
-                              SLEN(GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS),
-                              0};
-    int i = 0;
     uint32_t bitfield = 0;
 
     if (!xdata)
         return _gf_false;
 
     if (dict_del_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT))
-        SET_BIT(bitfield, GLUSTERFS_ENTRYLK_COUNT_BIT);
+        set_bit(&bitfield, GLUSTERFS_ENTRYLK_COUNT_BIT);
 
     if (dict_del_sizen(xdata, GLUSTERFS_INODELK_COUNT))
-        SET_BIT(bitfield, GLUSTERFS_INODELK_COUNT_BIT);
+        set_bit(&bitfield, GLUSTERFS_INODELK_COUNT_BIT);
 
     if (dict_del_sizen(xdata, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS))
-        SET_BIT(bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT);
+        set_bit(&bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT);
 
-    if (dict_del_sizen(xdata, GLUSTERFS_INODELK_DOM_COUNT))
-        SET_BIT(bitfield, GLUSTERFS_INODELK_DOM_COUNT_BIT);
+    if (dict_getn(xdata, GLUSTERFS_INODELK_DOM_COUNT,
+                  SLEN(GLUSTERFS_INODELK_DOM_COUNT)))
+        set_bit(&bitfield, GLUSTERFS_INODELK_DOM_COUNT_BIT);
 
-    if (dict_getn(xdata, GLUSTERFS_POSIXLK_COUNT,
-                  SLEN(GLUSTERFS_POSIXLK_COUNT)))
-        SET_BIT(bitfield, GLUSTERFS_POSIXLK_COUNT_BIT);
+    if (dict_del_sizen(xdata, GLUSTERFS_POSIXLK_COUNT))
+        set_bit(&bitfield, GLUSTERFS_POSIXLK_COUNT_BIT);
 
     if (dict_del_sizen(xdata, GLUSTERFS_PARENT_ENTRYLK))
-        SET_BIT(bitfield, GLUSTERFS_PARENT_ENTRYLK_BIT);
+        set_bit(&bitfield, GLUSTERFS_PARENT_ENTRYLK_BIT);
 
     return bitfield;
 }
@@ -229,7 +221,7 @@ dict_delete_domain_key(dict_t *dict, char *key, data_t *value, void *data)
 }
 
 static void
-pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, uint32_t bitfield)
+pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
 {
     if (!local || !xdata)
         return;
@@ -237,7 +229,8 @@ pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, uint32_t bitfield)
     GF_ASSERT(local->xdata == NULL);
     local->xdata = dict_copy_with_ref(xdata, NULL);
 
-    if (IS_BIT_SET(bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT)) {
+    if (IS_BIT_SET(local->bitfield,
+                   GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT)) {
         dict_foreach_fnmatch(xdata, GLUSTERFS_INODELK_DOM_PREFIX "*",
                              dict_delete_domain_key, NULL);
     }

